src/components/pages/packages/flyouts/packageNew/packageNew.js

```diff
@@ -52,23 +55,40 @@ export class PackageNew extends LinkedComponent {
   apply = (event) => {
     event.preventDefault();
     const { createPackage } = this.props;
-    const { type, packageFile } = this.state;
+    const { packageType, configType, customConfigName, packageFile } = this.state;
+
+    // If configType is 'Custom' concatenate 'Custom' with customConfigName.
+    let configName = '';
+    if (configType === configTypeOptions[1]) configName = `${configTypeOptions[1]} - ${customConfigName}`;
+    else configName = configType;
+
     this.props.logEvent(
       toDiagnosticsModel(
         'NewPackage_Apply',
         {
-          type,
+          packageType,
           packageName: packageFile.name
         })
     );
     if (this.formIsValid()) {
-      createPackage({ type: type, packageFile: packageFile });
-      this.setState({ changesApplied: true });
+      createPackage({ packageType: packageType, configType: configName, packageFile: packageFile });
```

Comment:
So we're saving a customizable string as a Configuration Type? Its not a separate column? Um, that feels wrong to me. Something referred to as a "Type" should have a set of known values. Otherwise, its just a "Name" (and should be called that instead of mixing semantics).

Reply:
This was thought through with PM. I'll explain in person.